src/Microsoft.DotNet.TemplateLocator/TemplateLocator.cs

```diff
@@ -5,68 +5,106 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
-using NuGet.Common;
-using NuGet.Protocol;
+using Microsoft.DotNet.DotNetSdkResolver;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
 
 namespace Microsoft.DotNet.TemplateLocator
 {
     public sealed class TemplateLocator
     {
-        private DirectoryInfo _dotnetSdkTemplatesLocation;
+        private readonly Lazy<NETCoreSdkResolver> _netCoreSdkResolver;
 
         public TemplateLocator()
+            : this(Environment.GetEnvironmentVariable, VSSettings.Ambient)
         {
-            string mockTemplateLocation = Environment.GetEnvironmentVariable("MOCKDOTNETSDKTEMPLATESLOCATION");
-            if (!string.IsNullOrWhiteSpace(mockTemplateLocation))
-            {
-                _dotnetSdkTemplatesLocation = new DirectoryInfo(mockTemplateLocation);
-            }
         }
 
-        public IReadOnlyCollection<IOptionalSdkTemplatePackageInfo> GetDotnetSdkTemplatePackages(string sdkVersion)
+        // Test constructor
+        public TemplateLocator(Func<string, string> getEnvironmentVariable, VSSettings vsSettings)
+        {
+            _netCoreSdkResolver =
+                new Lazy<NETCoreSdkResolver>(() => new NETCoreSdkResolver(getEnvironmentVariable, vsSettings));
+        }
+
+        public IReadOnlyCollection<IOptionalSdkTemplatePackageInfo> GetDotnetSdkTemplatePackages(
+            string sdkVersion,
```

Comment:
if either param is null, the result = unhandled exception.  Is this by design?

Reply:
good catch